dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoryHelper.java

```diff
@@ -54,4 +63,109 @@ public CategoryView toCategoryView(final Category category, final User user) {
                 .categoryVelocityVarName(category.getCategoryVelocityVarName())
                 .build();
     }
+
+    public void addOrUpdateCategory(final User user, final String contextInode,final BufferedReader br,final Boolean merge) throws IOException, Exception {
+
+        CsvReader csvreader = null;
+        try{
+            csvreader = new CsvReader(br);
```

Comment:
I am wondering if the import may be used in another context, I am not sure about it, but if it is we could encapsulate the CsvReader logic in a Class Such as 

for (final CategoryDTO categoryDTO : CategoryImporter.from(br)) // .iterable()
 addOrUpdateCategory(user, true, contextInode, categoryDTO.getInode(), categoryDTO.getName(),
                            categoryDTO.getVariable(), null, categoryDTO.getSort(), merge);

Reply:
This change has been addressed